src/librbd/image/RefreshRequest.cc

```diff
@@ -1445,7 +1451,8 @@ bool RefreshRequest<I>::get_migration_info(ParentImageInfo *parent_md,
                                            MigrationInfo *migration_info) {
   if (m_migration_spec.header_type != cls::rbd::MIGRATION_HEADER_TYPE_DST ||
       (m_migration_spec.state != cls::rbd::MIGRATION_STATE_PREPARED &&
-       m_migration_spec.state != cls::rbd::MIGRATION_STATE_EXECUTING)) {
+       m_migration_spec.state != cls::rbd::MIGRATION_STATE_EXECUTING &&
+       m_migration_spec.state != cls::rbd::MIGRATION_STATE_ABORTING)) {
```

Comment:
Just wondering what would happen if the aborting state was set by a new client, but  the refresh was running by an older client, which did not have this change. It looks like it would crash on the failed assert below?

On the other hand it looks like impossible because we check now the image is not currently used when setting the "aborting" state. So I assume `RefreshRequest<I>::get_migration_info` is never expected to be called if the current state is `aborting`? I.e. no need in this condition check, though it does not hurt.

Reply:
I think currently it would get stuck in a loop in `handle_get_migration_header` since it isn't one of the recognized types. I could refactor that state machine loop and `get_migration_info` to return an error code instead of aborting.

Reply:
Yes. I was just "future proofing" it in case it needs to change again.